src/plugins/elasticsearch/lib/create_agent.js

```diff
@@ -19,7 +19,7 @@ module.exports = _.memoize(function (server) {
   }
 
   // Add client certificate and key if required by elasticsearch
-  if (config.get('elasticsearch.ssl.cert') && config.get('elasticsearch.ssl.key')) {
+  if (config.get('elasticsearch.ssl.enabled')) {
```

Comment:
Should be accessing `server.ssl.enabled` since the config is set in the server schema rather than in https://github.com/elastic/kibana/blob/master/src/plugins/elasticsearch/index.js#L20-L25 ?